pkg/routes/updates.go

```
@@ -148,115 +149,125 @@ func GetUpdates(w http.ResponseWriter, r *http.Request) {
+	ctxServices := dependencies.ServicesFromContext(r.Context())
+	ctxServices.Log.Info("Update is being created")
+	account := readAccount(w, r, ctxServices.Log)
+	if account == "" {
+		// errors handled by readAccount
+		return nil
 	}
 
 	var devicesUpdate models.DevicesUpdate
-	err = json.NewDecoder(r.Body).Decode(&devicesUpdate)
-	if err != nil {
-		err := errors.NewBadRequest("Invalid JSON")
-		w.WriteHeader(err.GetStatus())
-		return nil, err
+	if err := readRequestJSONBody(w, r, ctxServices.Log, &devicesUpdate); err != nil {
+		return nil
 	}
-	services.Log.WithField("updateJSON", devicesUpdate).Debug("Update JSON received")
+	ctxServices.Log.WithField("updateJSON", devicesUpdate).Debug("Update JSON received")
 
 	// TODO: Implement update by tag - Add validation per tag
 	if devicesUpdate.DevicesUUID == nil {
-		err := errors.NewBadRequest("DeviceUUID required.")
-		w.WriteHeader(err.GetStatus())
-		return nil, err
+		respondWithAPIError(w, ctxServices.Log, errors.NewBadRequest("DeviceUUID required."))
+		return nil
+	}
+	// remove any duplicates
+	devicesUUID := make([]string, 0, len(devicesUpdate.DevicesUUID))
+	devicesUUIDSMap := make(map[string]bool, len(devicesUpdate.DevicesUUID))
+	for _, deviceUUID := range devicesUpdate.DevicesUUID {
+		if _, ok := devicesUUIDSMap[deviceUUID]; !ok {
+			devicesUUID = append(devicesUUID, deviceUUID)
+			devicesUUIDSMap[deviceUUID] = true
+		}
+	}
+	// check that all submitted devices exists
+	var devicesCount int64
+	if result := db.DB.Debug().Model(&models.Device{}).Where("account = ? AND uuid IN (?)", account, devicesUUID).Count(&devicesCount); result.Error != nil {
+		ctxServices.Log.WithFields(log.Fields{
+			"error": result.Error.Error(),
+		}).Error("failed to get devices count")
+		apiError := errors.NewInternalServerError()
+		apiError.SetTitle("failed to get devices count")
+		respondWithAPIError(w, ctxServices.Log, apiError)
+		return nil
+	}
+	if int64(len(devicesUUID)) != devicesCount {
```

I think we should log the error for the device not found but continue the process to the existent devices. In case we fire the update for 1000 devices and 10 are wrong, we should update the 990.
What do you think?

---

Think we should not silently ignore the non existent one, only in case we received that option from the user, say there is an option in the request to Ignore if not  exist, this may be realized a standard feature in our app, on how to deal with this situations. 
say three options:
0 - See the operation as atomic one, all items must pass or no one, as a default option.
1 - give option to ignore the non existent items.
2 - silently ignore the non existent one , but return back a list of the non existent items.

But think  anyway this is not a blocker for this (option 0) bug fix, as this is a feature  that should be discussed to be followed in other areas.